pkg/kube/container_oci.go

```diff
@@ -322,3 +321,50 @@ func (t *containerTranslator) configureAnnotations() {
 		t.g.AddAnnotation(k, v)
 	}
 }
+
+func (t *containerTranslator) configureUser() error {
+	security := t.cont.GetLinux().GetSecurityContext()
+	var userParts []string
+	if security.GetRunAsUsername() != "" {
+		userParts = append(userParts, security.GetRunAsUsername())
+	}
+	if security.GetRunAsUser() != nil {
+		userParts = append(userParts, fmt.Sprintf("%d", security.GetRunAsUser().GetValue()))
+	}
+	if security.GetRunAsGroup() != nil {
+		userParts = append(userParts, fmt.Sprintf("%d", security.GetRunAsGroup().GetValue()))
+	}
+
+	userSpec := strings.Join(userParts, ":")
+	containerUser, err := getContainerUser(t.cont.rootfsPath(), userSpec)
+	if err != nil {
+		return err
+	}
+
+	t.g.SetProcessUID(uint32(containerUser.Uid))
+	t.g.SetProcessGID(uint32(containerUser.Gid))
+	for _, gid := range containerUser.Sgids {
```

Comment:
Does it take groups associated with user from container `/etc/group` ?

Reply:
Yes. `containerUser` is already a valid user. Upon validation container's `/etc/passwd` and `/etc/groups` are read to make sure user & group exist.